### PR TITLE
Improvements to provisioning scripts

### PR DESCRIPTION
**Pull request by ProjectsByJackHe**
1. Added support to skip creation of VMs if they already exist.
2. Modified setup linux script to work on the experimental boost 4 VMs.

### CODE REVIEW

File: setup-runner-linux.sh [hunk 1]
```diff
@@ -47,12 +47,8 @@ if [[ -z "$username" ]]; then
   exit 1
 fi
 
-if [[ -z "$noreboot" ]]; then
-  noreboot=false
-fi
-
 if [[ -z "$runnerlabels" ]]; then
-  runnerlabels="experimental-ubuntu"
+  runnerlabels="azure-ex"
```
Comment (nibanks) [inline setup-runner-linux.sh:51]: Make this an input argument

File: .github/workflows/create-azure-machines.ps1 [hunk 2]
```diff
@@ -65,7 +68,11 @@ if ($SubscriptionId) {
     Connect-AzAccount -SubscriptionId $subscriptionId
 }
 
-Write-Host "Creating Azure Resources ($ResourceGroupName, $Location, $os, $VMSize)"
+if ($VMAlreadyExists) {
```
Comment (nibanks) [inline .github/workflows/create-azure-machines.ps1:71]: The first part of `Add-NetPerfVm` already checks and skips if it's there.

File: setup-runner-linux.sh [hunk 3]
```diff
@@ -71,19 +67,21 @@ echo "================= Configuring SSH. ================="
...
-  ssh-keygen -t rsa -b 2048 -N "" -f $HOME/.ssh/id_rsa
-  sshpass -p $password ssh-copy-id -o StrictHostKeyChecking=no $username@$peerip
+  ssh-keygen -t rsa -N "" -f $HOME/.ssh/id_rsa
+  sudo sshpass -p $password ssh-copy-id -i $HOME/.ssh/id_rsa.pub -o StrictHostKeyChecking=no $username@$peerip
+  chmod 777 $HOME/.ssh/id_rsa
   sudo systemctl restart ssh
 fi
 
-# Installing powershell 7
+# Installing powershell 7 (NOTE: Sometimes, will have to run this again because powershell fails to install the first run. Need investigation.)
```
Comment (nibanks) [inline setup-runner-linux.sh:84]: Put it in a loop, and keep running if `pwsh` doesn't work?

File: .github/workflows/create-azure-machines.ps1 (hunk 2 above)
Comment (ProjectsByJackHe) [reply .github/workflows/create-azure-machines.ps1:71]: You're right, this isn't needed.

File: setup-runner-linux.sh (hunk 1 above)
Comment (ProjectsByJackHe) [reply setup-runner-linux.sh:51]: It is an input argument. This checks if the user has specified anything. If not, assign it a default value of "azure-ex."

Comment (nibanks) [reply setup-runner-linux.sh:51]: Oh, I see. Ignore me then. 😄

File: setup-runner-linux.sh (hunk 3 above)
Comment (ProjectsByJackHe) [reply setup-runner-linux.sh:84]: Great idea, I wonder if I have to refresh my shell or bash does something weird... A follow PR is in the books.

**Pull request merged**